scipy/special/cephes/bdtr.c

```diff
@@ -148,103 +148,89 @@
 
 #include "mconf.h"
 
-double bdtrc(k, n, p)
-int k, n;
-double p;
+double bdtrc(double k, double n, double p)
 {
     double dk, dn;
 
     if (npy_isnan(p)) {
 	return NPY_NAN;
     }
-    if ((p < 0.0) || (p > 1.0)) {
-	goto domerr;
+    if (p < 0.0 || p > 1.0 || n < k) {
+	sf_error("bdtrc", SF_ERROR_DOMAIN, NULL);
+	return NPY_NAN;
     }
     if (k < 0) {
 	return 1.0;
     }
-
-    if (n < k) {
-      domerr:
-	sf_error("bdtrc", SF_ERROR_DOMAIN, NULL);
-	return NPY_NAN;
-    }
-
-    if (k == n)
-	return (0.0);
-    dn = n - k;
+    dn = floor(n) - floor(k);
     if (k == 0) {
 	if (p < .01)
 	    dk = -expm1(dn * log1p(-p));
 	else
 	    dk = 1.0 - pow(1.0 - p, dn);
     }
     else {
-	dk = k + 1;
+	dk = floor(k) + 1;
 	dk = incbet(dk, dn, p);
     }
-    return (dk);
+    return dk;
 }
 
 
 
-double bdtr(k, n, p)
-int k, n;
-double p;
+double bdtr(double k, double n, double p)
 {
     double dk, dn;
 
-    if ((p < 0.0) || (p > 1.0))
-	goto domerr;
-    if ((k < 0) || (n < k)) {
-      domerr:
-	sf_error("bdtr", SF_ERROR_DOMAIN, NULL);
-	return (NPY_NAN);
+    if (p < 0.0 || p > 1.0 || k < 0 || n < k) {
+        sf_error("bdtr", SF_ERROR_DOMAIN, NULL);
+        return NPY_NAN;
     }
 
     if (k == n)
-	return (1.0);
+	return 1.0;
 
-    dn = n - k;
+    dn = floor(n) - floor(k);
```

Comment:
As above about rounding `n`.

Reply:
see my question above. I'm not clear what you mean by the depracation comment.